src/iris/app_stats.py

```diff
@@ -173,7 +173,8 @@ def calculate_global_stats(connection, cursor, fields_filter=None):
                                                         WHERE (SELECT @row_id := @row_id + 1)
                                                         BETWEEN @incident_count/2.0 AND @incident_count/2.0 + 1)''',
         'total_applications': 'SELECT COUNT(*) FROM `application` WHERE `auth_only` = FALSE',
-        'high_priority_incidents_last_2_weeks': 'SELECT COUNT(DISTINCT incident.id) FROM incident JOIN message ON message.incident_id = incident.id WHERE incident.created > NOW() - INTERVAL 2 WEEK AND priority_id IN (SELECT id FROM priority WHERE name IN ("high","urgent"))'
+        'total_high_priority_incidents_last_2_weeks': 'SELECT COUNT(DISTINCT incident.id) FROM incident JOIN message ON message.incident_id = incident.id WHERE incident.created > NOW() - INTERVAL 2 WEEK AND priority_id IN (SELECT id FROM priority WHERE name IN ("high","urgent"))',
+        'high_priority_incidents_last_2_weeks_by_app': 'SELECT application.name, COUNT(DISTINCT incident.application_id, incident.id) FROM incident JOIN application ON application.id = incident.application_id JOIN message ON message.incident_id = incident.id WHERE incident.created > NOW() - INTERVAL 2 WEEK AND priority_id IN (SELECT id FROM priority WHERE name IN ("high","urgent")) GROUP BY incident.application_id'
```

Comment:
Shouldn't this go in `calculate_app_stats()`, right above?

Reply:
Depends on how many different applications are creating incidents we care about and what workflow we expect for consuming this stat. Moving it as per our discussion.